profile.d/meson.build

```diff
@@ -4,15 +4,33 @@ install_data('70-systemd-shell-extra.sh', install_dir : shellprofiledir.startswi
 install_data('80-systemd-osc-context.sh', install_dir : shellprofiledir.startswith('/usr/') ? shellprofiledir : libexecdir / 'profile.d')
 
 if conf.get('LINK_SHELL_EXTRA_DROPIN') == 1
-        install_emptydir(shellprofiledir)
+        if meson.version().version_compare('<1.3.0') # FATAL_WARNING_MARKER_REMOVE_SINGLE
```

Comment:
I think `>=1.3.0` (with the `else` clause handling `<1.3.0`) should do the trick without markers.

Reply:
interesting... thank you.

Reply:
This happens because meson knows how to special case the `meson.version().version_compare` function object, and when entering an if stack, it temporarily overrides the "effective" minimum version for the duration of the if stack.

So, inside of this block, it is ***as if*** you had a project `meson_version: '<1.3.0'`, which does not actually help avoid "feature too new" warnings.

In principle there's no reason we would turn down code that made this handle else blocks too. But it's not immediately apparent how to do so, since else blocks would have to understand the context of the original if, as well as elif statements, and invert the former reliably. (?) In nearly all cases you can just rewrite it using >= instead, which is simple to get correct.

Reply:
> In nearly all cases you can just rewrite it using >= instead, which is simple to get correct.

I think the problem is that it isn't obvious. For example ea07a6d45daf50d948a8a904f0417cf25c97ab65 was merged with `<1.3.0` and since that part has never been covered by the workflows with warnings it wasn't noticed. I think ideally it would be great if it worked without having to remember that `>=` should be used.

Reply:
I refer you back to

> In principle there's no reason we would turn down code that made this handle else blocks too. But it's not immediately apparent how to do so

Or as they say in the old country,

> Patches welcome :)

Reply:
(I really do mean this quite sincerely. I tried to figure out a way and failed. I wish it worked.)